api.md

```diff
@@ -390,6 +391,242 @@ if you try to upload another file other than an image:
 Or an image size that's too large,
 you'll get an `413 Request Entity Too Large` error.
 
+# 5. Returning meaningful errors
 
+If something *fails*, the person using the API
+will have a `JSON` body stating 
+`"Error uploading file #26"` returned.
+This type of information is not particularly *useful*.
```

Comment:
having an error that has a code e.g: `#26` (in this case a line number)
that the support staff or engineers can _easily_ identify and address
is orders of magnitude more _useful_ than `{:error, :upload_fail}`.

Reply:
I've added logging and pattern-matched the relevant information, all the while being compatible with the other feedback you've given (showing a generic error message to the person).